### PR TITLE
Fix bug in absolute timelock usage

### PR DESCRIPTION
**Pull request by tcharding**
Currently if we mix up height/time absolute timelocks when filtering policies the result is incorrect (to the best of my understanding).

Add a `timelock` module that includes a single public function for checking if absolute timelocks are the same unit. Use the new function to fix a bug in policy filtering.

- Patch 1 is a docs bug fix
- Patch 2 is the bug fix
- Patch 3 is a unit test patch that fails if its moved before patch 2. Please review the unit test carefully to make sure I'm not confused.

## Note

There is [ongoing discussion](https://github.com/rust-bitcoin/rust-bitcoin/pull/994) around trying to design a suitable timelock API. This PR is an attempt to make some forward progress by taking baby steps _and_ making objective improvements. I decided to do it here in miniscript because it will be easier to iterate on and more obvious when there are concrete usage examples along with each change.

cc dpc

### CODE REVIEW

File: src/timelock.rs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+//! Various functions for manipulating Bitcoin timelocks.
+
+use crate::miniscript::limits::LOCKTIME_THRESHOLD;
+
+/// Returns true if `a` and `b` are the same unit i.e., both are block heights or both are UNIX
+/// timestamps. `a` and `b` are nLockTime values.
+pub fn absolute_timelocks_are_same_unit(a: u32, b: u32) -> bool {
+    n_lock_time_is_block_height(a) && n_lock_time_is_block_height(b)
```
Comment (dpc) [inline src/timelock.rs:8]: I wonder if `n_lock_time_is_block_height(a) == n_lock_time_is_block_height(b)` is more readable to other people, or just me.

Comment (tcharding) [reply src/timelock.rs:8]: I like it!

**Comment by tcharding** [comment]
Force push is `cargo fmt` and suggestion from @dpc.

**Comment by tcharding** [comment]
Oh and I made the helper functions public because clippy warns otherwise, I think there is value in having both even though one is now not used. I specifically put the `<` and `>=` there, explicitly in front of everyone that sees it, instead of negating a function call in one to the other.

**Comment by sanket1729** [comment]
Thanks for this, will catch up the lock time discussions upstream this week

**Comment by sanket1729** [comment]
Edited the description to remove the @ mention

**Review by sanket1729 — APPROVED** [review]
ACK 42e12168f4c4bcb6f521016930fa6a92289631fd. 

This is clean improvement. A nice step towards #270

**Pull request merged**